tests/unit/modules/test_rbenv.py

```diff
@@ -76,6 +76,49 @@ def test_install_ruby(self):
                                       return_value=None):
                         self.assertFalse(rbenv.install_ruby('ruby'))
 
+    def test_install_ruby_with_string_config(self):
+        '''
+        Test for install a ruby implementation using rbenv:build_env as a string.
+        '''
+        with patch.dict(rbenv.__grains__, {'os': 'FreeBSD'}):
+            string_config_ret = 'RUBY_CFLAGS="-O3"'
+            with patch.dict(rbenv.__salt__,
+                            {'config.get': MagicMock(return_value=string_config_ret)}):
+                with patch.object(rbenv, '_rbenv_exec',
+                                  return_value={'retcode': 0,
+                                                'stderr': 'stderr'}):
+                    with patch.object(rbenv, 'rehash', return_value=None):
+                        self.assertEqual(rbenv.install_ruby('ruby'), 'stderr')
+
+                with patch.object(rbenv, '_rbenv_exec',
+                                  return_value={'retcode': 1,
+                                                'stderr': 'stderr'}):
+                    with patch.object(rbenv, 'uninstall_ruby',
+                                      return_value=None):
+                        self.assertFalse(rbenv.install_ruby('ruby'))
+
+    def test_install_ruby_with_dictionary_config(self):
+        '''
+        Test for install a ruby implementation using rbenv:build_env as a dictionary.
+        '''
+        with patch.dict(rbenv.__grains__, {'os': 'FreeBSD'}):
+            dict_config_ret = {'RUBY_CFLAGS': '-O3'}
+            with patch.dict(rbenv.__salt__,
+                            {'config.get': MagicMock(return_value=dict_config_ret)}):
+                with patch.object(rbenv, '_rbenv_exec',
+                                  return_value={'retcode': 0,
+                                                'stderr': 'stderr'}):
+                    with patch.object(rbenv, 'rehash', return_value=None):
+                        self.assertEqual(rbenv.install_ruby('ruby'), 'stderr')
+
+                with patch.object(rbenv, '_rbenv_exec',
+                                  return_value={'retcode': 1,
+                                                'stderr': 'stderr'}):
+                    with patch.object(rbenv, 'uninstall_ruby',
+                                      return_value=None):
+                        self.assertFalse(rbenv.install_ruby('ruby'))
+
+
```

Comment:
I have pretty much zero idea what I am doing in here. My intention was to test the different config options the install_ruby function could handle, along with the default.  These tests past on my local environment, but I don't know if there is a better way to do what I was trying to do here. So I just copied the original test and changed the mock data to fit the different kinds of config options the method could take.
Open to suggestions.